src/main/java/org/reaktivity/nukleus/http/internal/routable/stream/SourceInputStreamFactory.java

```diff
@@ -592,6 +695,34 @@ private void beforeWindowOrReset(
             }
         }
 
+        private void processInitialWindow(DirectBuffer buffer, int index, int length)
+        {
+            windowRO.wrap(buffer, index, index + length);
+            int update = windowRO.update();
+            doSourceWindow(update);
+
+            // Process any deferred source data
+            int dataLength = Math.min(slotPosition, update);
+            MutableDirectBuffer data = slab.buffer(slotIndex);
+            decode(data, slotOffset, dataLength);
+            if (dataLength < slotPosition)
+            {
+                slotOffset += dataLength;
+            }
+            else
+            {
+                slab.release(slotIndex);
+                slotIndex = SLOT_NOT_AVAILABLE;
+                if (endRequested)
+                {
+                    doEnd();
+                    return;
```

Comment:
We've tried to keep the methods as single return, so recommend an else clause here instead of a short-circuit return.